lib/statsig/configs.ex

```diff
@@ -0,0 +1,114 @@
+defmodule Statsig.Configs do
+  use GenServer
+  require Logger
+
+  @table_name :statsig_configs
+
+  defmodule State do
+    @type t :: %__MODULE__{
+            last_sync_time: integer(),
+            reload_timer: reference() | nil,
+            reload_interval: integer()
+          }
+
+    defstruct last_sync_time: 0,
+              reload_timer: nil,
+              reload_interval: 10_000
+
+    def new() do
+      %__MODULE__{reload_interval: reload_interval()}
+    end
+
+    def schedule_reload(%__MODULE__{} = state) do
+      timer = Process.send_after(self(), :reload_configs, state.reload_interval)
+      %__MODULE__{state | reload_timer: timer}
+    end
+
+    defp reload_interval() do
+      Application.get_env(:statsig, :config_reload_interval, 10_000)
+    end
+  end
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  def lookup(name, type) do
+    :ets.lookup(@table_name, {name, type})
+  end
+
+  @impl true
+  def init(_) do
+    :ets.new(@table_name, [:named_table, :set, :public])
+
+    state =
+      State.new()
+      |> State.schedule_reload()
+
+    case attempt_reload(state) do
+      {:ok, updated_state} -> {:ok, updated_state}
+      {:error, _reason, error_state} -> {:ok, error_state}
+    end
+  end
+
+  @impl true
+  def handle_info(:reload_configs, state) do
+    new_state = State.schedule_reload(state)
```

Comment:
you might want to put the schedule reload after you do the reload, since if you do it before, since the time taken to reload will effectively subtract from the reload cadence. 

Not a huge deal.